tests/unit/client.py

```diff
@@ -247,10 +247,41 @@ def avoid_network_race_condition():
     db_file = db_dir+'/haas.db'
     conn = sqlite3.connect(db_file)
     cur = conn.cursor()
+    timeout = 0
     while True:
         que = len(cur.execute('select * from networking_action;').fetchall())
         if que > 0:
             time.sleep(1)
+            timeout = timeout + 1
+            if timeout > 10:  # breakout if exceeds 10 secs.
+                break
+        else:
+            break
+
+
+def avoid_race_sqlalchemy():
+    from haas.model import Node, Project, NetworkingAction
+    from sqlalchemy import func
+    from sqlalchemy import create_engine
+    from sqlalchemy.orm import sessionmaker
```

Comment:
Always put imports at the top of the file, unless there's a *strong* reason not to (e.g. we can't do this with extensions in the tests, since otherwise the tests pollute each other). If you're importing stuff inside of a function, it means you're finding out about related errors later, which is generally a bad thing. It also makes it harder to understand what code you're actually pulling in, and when.

There are plenty of linters out there that can tell us about unused imports. One of these days we should rig one up to the CI and enforce things that way. That will be a much more reliable way of making sure we remember to remove stuff.